src/template.cpp

```diff
@@ -7,14 +7,35 @@
 #include <plustache/template.hpp>
 
 #include <string>
-#include <boost/regex.hpp>
-#include <boost/algorithm/string/trim.hpp>
+#include <cctype>
+#include <algorithm>
 
 #include <plustache/plustache_types.hpp>
 #include <plustache/context.hpp>
 
 using namespace Plustache;
 
+
+// trim from start
+static inline std::string &ltrim(std::string &s) 
```

Comment:
can you add doxygen style comments here (similar to what the existing functions have) and explain the code a little as well. This single magic line is pretty cryptic to read.